src/core/StringX.h

```diff
@@ -28,6 +28,8 @@
 #include "../os/typedefs.h"
 #include "./PwsPlatform.h"
 
+extern void trashMemory(void *buffer, size_t length);
```

Comment:
Can you change this to
#include "Util.h"
?
extern statements like these are harder to maintain.